### PR TITLE
Scripts: use Python 3

### PR DESCRIPTION
**Pull request by TSC21**
Python 2 close to EOL. This is part of the update process throughout the PX4 ecosystem.

### CODE REVIEW

File: scripts/xacro.py [hunk 1]
```diff
@@ -694,16 +724,19 @@ def main():
...
-                   " |    This document was autogenerated by xacro from %-30s | " % args[0],
-                   " |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED  %-30s | " % "",
-                   " %s " % ('=' * 83)]]
+        banner = [
+            xml.dom.minidom.Comment(c) for c in [
+                " %s " %
+                ('=' * 83), " |    This document was autogenerated by xacro from %-30s | " %
+                args[0], " |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED  %-30s | " %
+                "", " %s " %
+                ('=' * 83)]]
```
Comment (julianoes) [inline scripts/xacro.py]: This is much uglier now.

**Review by julianoes — APPROVED** [review]

File: scripts/xacro.py (hunk 1 above)
Comment (TSC21) [reply scripts/xacro.py]: Well this is PEP8 taking the lead here. It's more tight in terms of length though.

Comment (TSC21) [reply scripts/xacro.py]: I am going to at least fix this line as it was more readable before.

**Comment by TSC21** [comment]
Now it's just a matter of waiting for Docker Hub to build the containers so we can use them.

**Comment by julianoes** [comment]
The build error on macOS is something new, and it happened when compiling Gazebo on Arch today for me as well:
```
/usr/bin/moc -DBOOST_ALL_NO_LIB -DBOOST_DATE_TIME_DYN_LINK -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_PROGRAM_OPTIONS_DYN_LINK -DBOOST_REGEX_DYN_LINK -DBOOST_SYSTEM_DYN_LINK -DBOOST_THREAD_DYN_LINK -DCessnaGUIPlugin_EXPORTS -DHAVE_OPENGL -DLIBBULLET_VERSION=0.0 -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NO_DEBUG "-DQT_TESTCASE_BUILDDIR=\"/home/julianoes/.cache/yay/gazebo/src/gazebo-10.1.0/build\"" -DQT_TESTLIB_LIB -DQT_WIDGETS_LIB -I/home/julianoes/.cache/yay/gazebo/src/gazebo-10.1.0/test/gtest/include -I/home/julianoes/.cache/yay/gazebo/src/gazebo-10.1.0/build -I/usr/include/libusb-1.0 -I/home/julianoes/.cache/yay/gazebo/src/gazebo-10.1.0 -I/usr/include/ignition/msgs1 -I/usr/include/ignition/transport4 -I/usr/include/ignition/fuel_tools1 -I/usr/include/ignition/common1 -I/usr/include/ignition/math4 -I/usr/include/sdformat-6.2 -I/usr/include/OGRE/RTShaderSystem -I/usr/include/OGRE -I/usr/include/OGRE/Terrain -I/usr/include/OGRE/Paging -I/usr/include/OGRE/Overlay -I/usr/include/qt -I/usr/include/qt/QtCore -I/usr/lib/qt/mkspecs/linux-g++ -I/usr/include/uuid -I/usr/include/qt/QtWidgets -I/usr/include/qt/QtGui -I/usr/include/qt/QtTest -I/usr/include -I/usr/include/c++/9.2.0 -I/usr/include/c++/9.2.0/x86_64-pc-linux-gnu -I/usr/include/c++/9.2.0/backward -I/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/include -I/usr/local/include -I/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/include-fixed --include /home/julianoes/.cache/yay/gazebo/src/gazebo-10.1.0/build/plugins/CessnaGUIPlugin_autogen/moc_predefs.h -p plugins -o /home/julianoes/.cache/yay/gazebo/src/gazebo-10.1.0/build/plugins/CessnaGUIPlugin_autogen/EWIEGA46WW/moc_CessnaGUIPlugin.cpp /home/julianoes/.cache/yay/gazebo/src/gazebo-10.1.0/plugins/CessnaGUIPlugin.hh

usr/include/tbb/tbb_profiling.:28: Parse error at "{"
```

**Comment by julianoes** [comment]
The CI errors are unrelated to Python 3. I'm therefore merging this.

**Review by julianoes — APPROVED** [review]

**Comment by dayjaby** [comment]
```
$ python3 --version
Python 3.6.9

$ python2 --version
Python 2.7.17
```

After this PR, I get this error message on a fresh ubuntu 18.04 install with python 2 and 3:

```CMake Error at /usr/share/cmake-3.10/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find PythonInterp: Found unsuitable version "2.7.17", but
  required is at least "3" (found /usr/bin/python)
```

**Comment by julianoes** [comment]
Do you have all dependencies as recommended below?

https://github.com/PX4/Firmware/blob/86bf4b2289c1630833c920c84dfc95a82e2ac6b5/Tools/setup/ubuntu.sh#L91-L95

**Pull request merged**